aci/resource_aci_fvnsmcastaddrblk.go

```diff
@@ -0,0 +1,219 @@
+	d.SetId(fvnsMcastAddrBlk.DistinguishedName)
+	d.Set("description", fvnsMcastAddrBlk.Description)
+	fvnsMcastAddrBlkMap, err := fvnsMcastAddrBlk.ToMap()
+	if err != nil {
+		return d, err
+	}
+	dn := d.Id()
+	if dn != fvnsMcastAddrBlk.DistinguishedName {
+		d.Set("multicast_pool_dn", "")
+	} else {
+		d.Set("multicast_pool_dn", GetParentDn(fvnsMcastAddrBlk.DistinguishedName, fmt.Sprintf("/"+models.RnfvnsMcastAddrBlk, fvnsMcastAddrBlkMap["from"], fvnsMcastAddrBlkMap["to"])))
+	}
+	d.Set("annotation", fvnsMcastAddrBlkMap["annotation"])
+	d.Set("from", fvnsMcastAddrBlkMap["from"])
+	d.Set("name", fvnsMcastAddrBlkMap["name"])
+	d.Set("name_alias", fvnsMcastAddrBlkMap["nameAlias"])
+	d.Set("to", fvnsMcastAddrBlkMap["to"])
+	return d, nil
+}
+
+func resourceAciMulticastAddressBlockImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+	fvnsMcastAddrBlk, err := getRemoteMulticastAddressBlock(aciClient, dn)
+	if err != nil {
+		return nil, err
+	}
+	schemaFilled, err := setMulticastAddressBlockAttributes(fvnsMcastAddrBlk, d)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciMulticastAddressBlockCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] MulticastAddressBlock: Beginning Creation")
+	aciClient := m.(*client.Client)
+	desc := d.Get("description").(string)
+	log.Printf("[DEBUG] MulticastAddressBlock: desc %s", desc)
+	from := d.Get("from").(string)
+	to := d.Get("to").(string)
+	MulticastAddressPoolDn := d.Get("multicast_pool_dn").(string)
+
+	fvnsMcastAddrBlkAttr := models.MulticastAddressBlockAttributes{}
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		fvnsMcastAddrBlkAttr.Annotation = Annotation.(string)
+	} else {
+		fvnsMcastAddrBlkAttr.Annotation = "{}"
+	}
+
+	if From, ok := d.GetOk("from"); ok {
+		fvnsMcastAddrBlkAttr.From = From.(string)
+	}
+
+	if Name, ok := d.GetOk("name"); ok {
+		fvnsMcastAddrBlkAttr.Name = Name.(string)
+	}
+
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		fvnsMcastAddrBlkAttr.NameAlias = NameAlias.(string)
+	}
+
+	if To, ok := d.GetOk("to"); ok {
+		fvnsMcastAddrBlkAttr.To = To.(string)
+	}
+	fvnsMcastAddrBlk := models.NewMulticastAddressBlock(fmt.Sprintf(models.RnfvnsMcastAddrBlk, from, to), MulticastAddressPoolDn, desc, fvnsMcastAddrBlkAttr)
+
+	err := aciClient.Save(fvnsMcastAddrBlk)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(fvnsMcastAddrBlk.DistinguishedName)
+	log.Printf("[DEBUG] %s: Creation finished successfully", d.Id())
+	return resourceAciMulticastAddressBlockRead(ctx, d, m)
+}
+func resourceAciMulticastAddressBlockUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
```

Comment:
Update functionality not working as expected.

Reply:
ok fixed